massgov/mayflower
# React/DP-12921: Prevents min/max from being past on InputNumber and InputCurrency.

## Description
* This changes InputNumber and InputCurrency so that changes made can't go beyond the passed in min/max values.
* This also adds the[ npm package "is"](https://www.npmjs.com/package/is) to the repo. We should use this going forward for type checking.
* Focusing InputCurrency will now remove the placeholder text if the input has no value, and place it back on blur if there is still no value.
* Finally, InputCurrency has been changed so that focusing no longer switches the currency formatting to numbers.

## Related Issue / Ticket

- [DP-12921](https://jira.mass.gov/browse/DP-12921)
- [DP-12734](https://jira.mass.gov/browse/DP-12734)

## Steps to Test
1. Pull the branch, npm install, run storybook, and browse to atoms -> forms -> Input(Number or Currency). Ensure that either input can no longer go beyond their set min or max values.
